R/forecastingdata_org.R

```diff
@@ -0,0 +1,380 @@
+  
+  data <- as.data.frame(matrix(nrow = row_count, ncol = length(col_names) + 1))
+  colnames(data) <- c(col_names, value_column_name)
+  
+  for(col in col_names)
+    data[[col]] <- eval(parse(text = col))
+  
+  data[[value_column_name]] <- values
+  
+  if(!(is.null(key))){
+    if(!(key %in% col_names))
+      stop("Invalid key. Cannot convert data into tsibble format.")
+    else{
+      if(is.null(index)){
+        if(is.null(index_var))
+          cat("Index is not provided. No valid index found in data. Returning a data.frame.")
+        else
+          data <- tsibble::build_tsibble(x = data, key = key, index = index_var, ordered = F)
+      }else{
+        if(!(index %in% col_names))
+          stop("Invalid index Cannot convert data into tsibble format.")
+        else
+          data <- tsibble::build_tsibble(x = data, key = key, index = index, ordered = F)
+      }
+    }
+  }else{
+    cat("Key is not provided. Returning a data.frame.")
+  }
+  
+  list(data, frequency, forecast_horizon, contain_missing_values, contain_equal_length)
+}
+
+
+
+#' download_data function
+#' 
+#' This function downloads a zip file from Zenodo platform given the record identifier and file name, unzip the contents and converts the unzipped .tsf file into a tsibble or a data.frame given the key and index.
+#' For more details, please refer to our paper: Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @param record_id Record identifier of the file that needs to be downloaded from Zenodo.
+#' @param file_name Name of the file (without extensions) that needs to be downloaded from Zenodo.
+#' @param destination_folder Folder path where the downloaded files need to be stored. By default, it uses the R working directory to store files.
+#' @param index The name of the time attribute that should be used as the index when creating the tsibble. If doesn't provide, it tries to find a valid index within the data. If there is no valid index, then a data frame will be returned instead of a tsibble.
+#' @param key The name of the attribute that should be used as the key when creating the tsibble. The default value is "series_name" which is the key of the datasets in our repository. If doesn't provide, a data frame will be returned instead of a tsibble.
+#' @param value_column_name Any name that is preferred to have as the name of the column containing series values in the returning tsibble.
+#'
+#' @return This function returns a tsibble or a data.frame based on the validity of given key and index values.
+#' @importFrom utils download.file unzip
+#' 
+#' @references 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Repository. \url{https://forecastingdata.org/}
+#' 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @export
```

Comment:
I don't think this should be exported, as Zenodo presumably hosts data which cannot (or should not) be converted to a tsibble.

Reply:
Fixed as suggested